random/README.md

```diff
@@ -0,0 +1,182 @@
+
+```
+Output:
+{
+  "Success": true,
+  "Error": "",
+  "Value": 3
+}
+```
+
+## Walkthrough
+
+### Step 1: Create a parameterized random function
+
+We'll demonstrate random by emulating a die roll in `rollDie` in
+[`main.go`](./main.go). As in previous examples, we will call this function
+using the `bky-as` CLI by passing in the
+[`fn-call.json`](./fn-call.json) file contents:
+
+```json
+{
+  "code_file": "tmp/x.wasm",
+  "function": "rollDie",
+  "input": {
+    "sides": 20
+  }
+}
+```
+
+Notice the `input` section, which contains the parameters for `rollDie`,
+specifically the `sides` field set to `20`. Meaning, `rollDie` will return
+a number in the range [1,20].
+
+Next, we define the `rollDie` function in [`main.go`](./main.go):
+
+```go
+type Args struct {
+    Sides uint64 `json:"sides"`
+}
+
+//export rollDie
+func rollDie(inputPtr uint64, secretPtr uint64) uint64 {
+    var input Args
+    inputData := basm.ReadFromHost(inputPtr)
+    err := json.Unmarshal(inputData, &input)
+    switch {
+    case err != nil:
+        outErr := fmt.Errorf("could not unmarshal input args: %w", err)
+        return WriteError(outErr)
+    case input.Sides == 0:
+        outErr := fmt.Errorf("die cannot have zero sides")
+        return WriteError(outErr)
+    }
+
+    randBytes := make([]byte, 8)
+    n, err := crand.Read(randBytes)
+    switch {
+    case err != nil:
+        outErr := fmt.Errorf("reading random bytes: %w", err)
+        return WriteError(outErr)
+    case n != 8:
+        outErr := fmt.Errorf("reading random bytes: expected 8 bytes, got %d", n)
+        return WriteError(outErr)
+    }
+
+    roll := (binary.BigEndian.Uint64(randBytes) % input.Sides) + 1
+    return WriteOutput(roll)
+}
+```
+
+You will notice a few things:
+
+- The `rollDie` function is exported so that it can be invoked by the
+  Blocky AS server in a WASM runtime.
+- The function takes two `uint64` arguments and returns a `uint64`. These are
+  fat pointers to shared memory managed by the Blocky AS server, where the first
+  32 bits are a memory address and the second 32 bits are the size of the data.
+  The memory space is sandboxed and shared between the TEE host program (Blocky
+  AS server) and the WASM runtime (your function). The `inputPtr` and
+  `secretPtr` arguments carry user-defined function input and secrets,
+  though we only make use of input in this example. The output of the function
+  is also a memory pointer, whose value will be returned to the user.
+- The function uses the `basm`
+  [Blocky Attestation Service WASM Go SDK](https://github.com/blocky/basm-go-sdk)
+  `basm.ReadFromHost` to fetch the input data and unmarshal it into the
+`Args` struct.
+
+Finally, note that we are able to generate random numbers the way you would in
+a normal Go program (i.e., via `crypto/rand`). This is because
+`tinygo` provides an implementation for `crypto/rand` and we properly seed the
```

Comment:
If you were familiar with go, you may wonder why we are using crypto/rand to generate a random dice roll and not `rand.Intn`.  Is it worth a sentence?

Reply:
Ahhh... Honestly, I was only using `crypto/rand` because I'd been using it recently for things that actually required cryptographically secure randomness.

I've updated the example to simply use `math/rand.Intn()` since emulating a die roll does not necessarily require `crypto/rand`. I've also updated these lines so that a reader is aware they could use either `math/rand` or `crypto/rand` depending on their use case